lms/static/scripts/frontend_apps/components/FilePickerFormFields.tsx

```diff
@@ -36,20 +45,34 @@ export default function FilePickerFormFields({
   title,
   content,
   formFields,
+  deepLinkingFields,
   groupSet,
 }: FilePickerFormFieldsProps) {
-  return (
-    <>
-      {Object.entries(formFields).map(([field, value]) => (
-        <input key={field} type="hidden" name={field} value={value} />
-      ))}
-      <input type="hidden" name="group_set" value={groupSet ?? ''} />
-      {content.type === 'url' && (
-        // Set the `document_url` form field which is used by the `configure_assignment`
-        // view. Used in LMSes where assignments are configured on first launch.
-        <input name="document_url" type="hidden" value={content.url} />
-      )}
-      {title !== null && <input type="hidden" name="title" value={title} />}
-    </>
-  );
+  // If we are deeplinking, ie submith the form to an LMS
+  // send the deepLinkingFields only.
+  if (deepLinkingFields) {
+    return (
```

Comment:
This is not very fancy code but I like the erxpclitness of it about deep linking vs not.

Of course I don't know how bad this looks in JS.